### PR TITLE
Print NullRestricted flag for ddr classAndFlags

### PR DESCRIPTION
**Pull request by theresa-m**
Related: https://github.com/eclipse-openj9/openj9/issues/17340

### CODE REVIEW

File: runtime/oti/j9javaaccessflags.h [hunk 1]
```diff
@@ -157,9 +157,7 @@
 #define J9StaticFieldRefBoolean 0x8
 #define J9StaticFieldRefPutResolved 0x10
 #define J9StaticFieldRefFinal 0x20
-#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
```
Comment (hangshao0) [inline runtime/oti/j9javaaccessflags.h:160]: As this ifdef is removed, can you add a comment above this flag saying that `J9StaticFieldIsNullRestricted` could only be set in Valhalla builds.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/CompatibilityConstants29.dat [hunk 2]
```diff
@@ -68,6 +68,8 @@ J9JavaClassFlags.J9ClassLargestAlignmentConstraintDouble = 0
 J9JavaClassFlags.J9ClassLargestAlignmentConstraintReference = 0
 J9JavaClassFlags.J9ClassRequiresPrePadding = 0
 
+J9JavaAccessFlags.J9StaticFieldIsNullRestricted = 0
```
Comment (keithc-ca) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/CompatibilityConstants29.dat:71]: Please move this into the existing group for `J9JavaAccessFlags`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: debugtools/DDR_VM/src/com/ibm/j9ddr/CompatibilityConstants29.dat (hunk 2 above)
Comment (theresa-m) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/CompatibilityConstants29.dat]: updated

File: runtime/oti/j9javaaccessflags.h (hunk 1 above)
Comment (keithc-ca) [reply runtime/oti/j9javaaccessflags.h:160]: Please explain the motivation for making this declaration unconditional (I only see uses where `defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)` is true).

Comment (theresa-m) [reply runtime/oti/j9javaaccessflags.h:160]: Wouldn't it not compile in the LinearDumper case? Or does DDR not pay attention to these?

Comment (keithc-ca) [reply runtime/oti/j9javaaccessflags.h:160]: No, it should compile: DDR benefits from the declaration of `J9StaticFieldIsNullRestricted` in `CompatibilityConstants29.dat`.

Comment (theresa-m) [reply runtime/oti/j9javaaccessflags.h:160]: Oh right, I'll put those back then.

**Comment by keithc-ca** [comment]
Jenkins test sanity aarch64_linux_vt_standard jdknext

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**